pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -228,6 +228,10 @@ func (l *clusterLifecycle) Updated(obj *v3.Cluster) (runtime.Object, error) {
 }
 
 func (l *clusterLifecycle) Remove(obj *v3.Cluster) (runtime.Object, error) {
+	if len(obj.Finalizers) > 1 {
+		return obj, errors.New("waiting to be the last finalizer")
```

Comment:
Can we use `generic.ErrSkip` instead so a log doesn't get out and we push retry to its max value.